src/arrayMethodReduce.js

```diff
@@ -5,7 +5,19 @@
  */
 function applyCustomReduce() {
   [].__proto__.reduce2 = function(callback, initialValue) {
-    // write code here
+    let prev = initialValue;
+    let startIndex = 0;
+
+    if (arguments.length < 2) {
+      prev = this[0];
+      startIndex = 1;
+    }
+
+    for (startIndex; startIndex < this.length; startIndex++) {
+      prev = callback(prev, this[startIndex], startIndex, this);
+    }
```

Comment:
it's better to use separate variable for Index
`for (let index = startIndex; index < this.length;  index++) {`

Reply:
Else ok 👍

Reply:
corrected